storage-bigtable/src/access_token.rs

```diff
@@ -34,12 +34,24 @@ fn load_stringified_credentials(credential: String) -> Result<Credentials, Strin
     Credentials::from_str(&credential).map_err(|err| format!("{err}"))
 }
 
-#[derive(Clone)]
-pub struct AccessToken {
+pub struct AccessTokenInner {
     credentials: Credentials,
     scope: Scope,
-    refresh_active: Arc<AtomicBool>,
-    token: Arc<RwLock<(Token, Instant)>>,
+    token: RwLock<(Token, Instant)>,
+    refresh_active: AtomicBool,
+}
+
+#[derive(Clone)]
+pub struct AccessToken {
+    inner: Arc<AccessTokenInner>,
+}
+
+impl std::ops::Deref for AccessToken {
+    type Target = AccessTokenInner;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
```

Comment:
Can you make this refactor a separate commit so that it's apparent why it's needed?